From 893949781d194032f7f490bd67fb4d39b7973ac3 Mon Sep 17 00:00:00 2001
From: Wei-Chiu Chuang <weichiu@apache.org>
Date: Tue, 30 Aug 2016 10:43:20 -0700
Subject: [PATCH 1931/2748] HDFS-10760. DataXceiver#run() should not log
 InvalidToken exception as an error. Contributed
 by Pan Yuxuan.

(cherry picked from commit c4ee6915a14e00342755d7cdcbf2d61518f306aa)
(cherry picked from commit 80bb9ffb67ef6e150e8dc4dd6afd2fcd0f823da5)

Change-Id: If0fb524d2e2abeb5914f4151790b600183d193f1
---
 .../hadoop/hdfs/server/datanode/DataXceiver.java   |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataXceiver.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataXceiver.java
index eba6c1e..93b04eb 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataXceiver.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataXceiver.java
@@ -269,6 +269,12 @@ public void run() {
         } else {
           LOG.info(s1 + "; " + t);          
         }
+      } else if (t instanceof InvalidToken) {
+        // The InvalidToken exception has already been logged in
+        // checkAccess() method and this is not a server error.
+        if (LOG.isTraceEnabled()) {
+          LOG.trace(s, t);
+        }
       } else {
         LOG.error(s, t);
       }
-- 
1.7.9.5

